File: checkoutsys-ui/checkoutsys.ui/src/app/models/transaction.ts
import { Order } from "./order";
import { User } from "./user";

export class Transaction {
    id: number = 0;
    orderId: number = 0;
    customerId: number = 0;
    date: Date = new Date();
    totalPrice: number = 0;
    customer?: User;
    order?: Order;
}

File: checkoutsys-api/checkoutsys.api/checkoutsys.api/Models/Transaction.cs
using System;
using System.Collections.Generic;

namespace checkoutsys.api.Models;

public partial class Transaction
{
    public long Id { get; set; }

    public long OrderId { get; set; }

    public long CustomerId { get; set; }

    public DateTime DateTime { get; set; }

    public decimal TotalPrice { get; set; }

    public virtual User Customer { get; set; } = null!;

    public virtual Order Order { get; set; } = null!;
}
